### ISSUE TITLE
Generate documentation for type aliases/definitions

### ISSUE THREAD

**Issue by bobogei81123** [rust-lang/rust#113727]
I'm facing similar problem to this [stack overflow question](https://stackoverflow.com/questions/74333290/how-to-generate-rust-docs-for-type-aliases).

Specifically, I made a wrapper class that wraps around an FFI object and creates common functionality like `from_ffi`, etc.

```rust
pub struct FfiWrapper<T>(T);

impl<T> FfiWrapper<T> {
  fn from_ffi(t: T) -> Self { ... }
}
```

Then, I implemented different methods based on the FFI type:

```rust
/// Some docs for type A
type A = FfiWrapper<FfiA>

impl A {
  /// Some docs for method_a
  fn method_a(self) ...
}
```

Yet, rust doc generate item and doc for `type A = FfiWrapper<FfiA>`, but not for `fn method_a`. How can I let the items in the specific implementation show up in docs?

**Comment by GuillaumeGomez** [comment]
You need to make `method_a` public.

**Comment by bobogei81123** [comment]
Hey, sorry for reporting to the wrong place.

I found out that what happened is that the wrapper is private but the type alias is public:
For example:

```
mod wrapper {
    #[repr(transparent)]
    pub struct Wrapper<T>(T);

    impl<T> Wrapper<T> {
        /// Common method
        pub fn common() {}
    }
}

pub mod a {
    use crate::wrapper::Wrapper;

    pub struct A;
    pub type WA = Wrapper<A>;

    impl WA {
        /// Common method
        pub fn method_a() {}
    }
}
```

The code compiles without any error and warning (See [rust playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a03b8712cd9eb540cd6fb6b574128b5e)). Yet, if you run `rust doc` then no doc will be generated for `Wrapper::common` and `WA::method_a`.

**Comment by GuillaumeGomez** [comment]
For the missing impl on the type alias, I made https://github.com/rust-lang/rust/pull/112429 but we needed to revert it because it introduced some bugs. We'll need to wait for the trait system rewrite to be done before I can put back the fix again.

**Comment by fmease** [comment]
Fixed by #115201, right?
